ansible/roles/jenkins/tasks/main.yml

```
@@ -45,6 +45,23 @@
   register: jenkins_password
   changed_when: jenkins_password.rc == 0
 
+- name: Install jenkins plugins
+  community.general.jenkins_plugin:
+    name: "{{ item }}"
+  register: jenkins_plugins
+  with_dict: "{{ jenkins.jenkins_plugins }}"
```

Since `jenkins.jenkins_plugins` is a list you should use `with_list`

```suggestion
  with_list: "{{ jenkins.jenkins_plugins }}"
```
Be aware that ` {{ jenkins }}` is a dict and  `{{ jenkins.jenkins_plugins }}` is a list